flink-streaming-connectors/flink-connector-hbase/src/main/java/org/apache/flink/streaming/connectors/hbase/HBaseMapper.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.hbase;
+
+import org.apache.flink.api.common.functions.Function;
+
+import java.io.Serializable;
+
+/**
+ * Maps a input value to a row in HBase table.
+ *
+ * @param <IN> input type
+ */
+public interface HBaseMapper<IN> extends Function, Serializable {
+
+	/**
+	 * Given an input value return the HBase row key. Row key cannot be null.
+	 *
+	 * @param value
+	 * @return row key
+	 */
+	byte[] rowKey(IN value);
```

Comment:
Hi @ramkrish86 , if we have 
MutationActions mutationActions  = HBaseMapper#actions(rowkey, value);
Then, the rowkey information would be included in mutationActions object being created. So when we call mutationActions. createMutations() there would be no need to pass rowkey as an argument, which I think is totally fine. But what I'm confused is how do we derive rowkey from the input value? Should we pass input value as an argument to getRowKey() API you mentioned? If so, getRowKey() API would be the same as rowkey(IN value) except a name change.